dht_bootstrap.go

```diff
@@ -41,52 +43,47 @@ func init() {
 	}
 }
 
-// BootstrapConfig runs cfg.Queries bootstrap queries every cfg.BucketPeriod.
-func (dht *IpfsDHT) Bootstrap(ctx context.Context) error {
-	triggerBootstrapFnc := func() {
-		logger.Infof("triggerBootstrapFnc: RT only has %d peers which is less than the min threshold of %d, triggering self & bucket bootstrap",
-			dht.routingTable.Size(), minRTBootstrapThreshold)
-
-		if err := dht.selfWalk(ctx); err != nil {
-			logger.Warningf("triggerBootstrapFnc: self walk: error: %s", err)
-		}
+// Bootstrap  i
+func (dht *IpfsDHT) startBootstrapping() error {
+	// scan the RT table periodically & do a random walk on k-buckets that haven't been queried since the given bucket period
+	dht.proc.Go(func(proc process.Process) {
```

Comment:
I figured I'd let the process management system manage the bootstrap process while we were at it.